src/main/java/uk/gov/ons/ctp/response/action/service/ProcessEventService.java

```diff
@@ -0,0 +1,766 @@
+      }
+    } else {
+      log.with("collectionExercise", collectionExercise)
+          .with("event", eventTag)
+          .info("No Emails to process");
+    }
+    log.debug("Getting Email cases against collectionExerciseId and event active enrolment");
+    List<ActionCase> letter_cases =
+        actionCaseRepository.findByCollectionExerciseIdAndActiveEnrolment(
+            collectionExerciseId, false);
+    if (letter_cases.size() > 0) {
+      ActionTemplate actionTemplate = actionTemplateService.mapEventTagToTemplate(eventTag, false);
+      if (actionTemplate != null) {
+        log.with("Letter cases", letter_cases.size()).info("Processing letter cases");
+        processLetterCases(letter_cases, actionTemplate, survey, collectionExercise, instant, true);
+      } else {
+        log.with("activeEnrolment", false)
+            .with("event", eventTag)
+            .error("No Template registered against the event and active enrolment");
+      }
+    } else {
+      log.with("collectionExercise", collectionExercise)
+          .with("event", eventTag)
+          .info("No Letters to process");
+    }
+  }
+
+  /**
+   * processes letter cases. populates letter entries for case parties which are actionable.
+   *
+   * @param letter_cases
+   * @param actionTemplate
+   * @param survey
+   * @param collectionExercise
+   */
+  private boolean processLetterCases(
+      List<ActionCase> letter_cases,
+      ActionTemplate actionTemplate,
+      SurveyDTO survey,
+      CollectionExerciseDTO collectionExercise,
+      Instant instant,
+      boolean isNotRetry) {
+    log.with("no. of cases", letter_cases.size())
+        .with("actionType", actionTemplate.getType())
+        .info("Populating letter data for letter cases.");
+    List<LetterEntry> letterEntries =
+        letter_cases
+            .parallelStream()
+            .filter(c -> isActionable(c, actionTemplate))
+            .map(c -> getPrintFileEntry(c, actionTemplate, survey))
+            .collect(Collectors.toList());
+    log.with("no. of cases", letter_cases.size())
+        .with("actionType", actionTemplate.getType())
+        .info("Finished populating letter data for letter cases.");
+    String fileNamePrefix =
+        FilenamePrefix.getPrefix(actionTemplate.getPrefix())
+            + "_"
+            + survey.getSurveyRef()
+            + "_"
+            + getExerciseRefWithoutSurveyRef(collectionExercise.getExerciseRef());
+    final String now =
+        DateTimeFormatter.ofPattern("ddMMyyyy_HHmm")
+            .withZone(ZoneId.systemDefault())
+            .format(instant);
+    String filename = String.format("%s_%s.csv", fileNamePrefix, now);
+    log.info("filename: " + filename + ", uploading file");
+    log.with("actionType", actionTemplate.getType()).info("Processing Print File");
+    boolean isSuccess = printFileService.processPrintFile(filename, letterEntries);
+    log.with("process file status?", isSuccess)
```

Comment:
Would reword this to "file processed?" just for a little more clarity when reading logs

Reply:
👍